### PR TITLE
Loki: Add error handling to `CallResource`

### PR DESCRIPTION
**Pull request by svennergr**
**What is this feature?**

This PR improves error handling in Loki's `CallResource` handler. Previously there were no errors surfaced whereas now there should be error popups like this:
image

**Which issue(s) does this PR fix?**:

Part of #64411

**Special notes for your reviewer**:

How to reproduce it (as minimally and precisely as possible):

1. Run Grafana in development mode
2. Select a Loki datasource
3. Set the timerange to a long range, like 2 years
4. Failed to call resource error appeared

### CODE REVIEW

**Comment by svennergr** [comment]
> Question: since we say it's an error, could it be that we're displaying it as a warning (yellow color):
> 
> image
> 
> vs
> 
> imagen

I did not change any frontend code in this PR, so I don't exactly know what's happening here. Also I'd like to keep this PR solely for backend changes and cover frontend topics in an other PR.

**Review by matyax — COMMENTED** [review]
Looks good to me, left one comment, in my first attempt to review Go code 😅

File: pkg/tsdb/loki/api.go [hunk 1]
```diff
@@ -97,13 +99,32 @@ func makeDataRequest(ctx context.Context, lokiDsUrl string, query lokiQuery) (*h
...
+	var data lokiError
+	err := json.Unmarshal(bytes, &data)
+	if err != nil {
+		// we were unable to convert the bytes to JSON, we return the whole text
+		return fmt.Errorf("%v", string(bytes))
+	}
+
+	errorMessage := data.Message
+
+	if errorMessage == "" {
```
[the remark this answers is not in the document]
Comment (svennergr) [reply pkg/tsdb/loki/api.go]: I was blindly copying from here https://github.com/grafana/grafana/pull/64439/files#diff-583c79313c9a2218e5aa7fa0fb37ba981a06839644491d89792f529ceefb1108L131-L139. Fixed in f755688

**Review by matyax — APPROVED** [review]
👌

**Review by matyax — APPROVED** [review]

**Pull request merged**